src/onboarding/steps.ts

```diff
@@ -0,0 +1,214 @@
+import { BIOMETRY_TYPE } from 'react-native-keychain'
+import { initializeAccount } from 'src/account/actions'
+import {
+  choseToRestoreAccountSelector,
+  recoveringFromStoreWipeSelector,
+} from 'src/account/selectors'
+import {
+  numberVerifiedCentrallySelector,
+  skipVerificationSelector,
+  supportedBiometryTypeSelector,
+} from 'src/app/selectors'
+import { setHasSeenVerificationNux } from 'src/identity/actions'
+import * as NavigationService from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { RootState } from 'src/redux/reducers'
+import { store } from 'src/redux/store'
+
+export const FIRST_ONBOARDING_SCREEN = Screens.NameAndPicture
+export const END_OF_ONBOARDING_SCREEN = Screens.WalletHome
+
+interface NavigatorFunctions {
+  navigate: typeof NavigationService.navigate | ((screen: Screens) => void)
+  popToScreen: typeof NavigationService.popToScreen | ((screen: Screens) => void)
+  navigateHome: typeof NavigationService.navigateHome | (() => void)
+  navigateClearingStack:
+    | typeof NavigationService.navigateClearingStack
+    | ((screen: Screens) => void)
+}
+
+interface GetStepInfoProps {
+  firstScreenInStep: Screens
+  navigator: NavigatorFunctions
+  dispatch: (action: any) => void
+  props: OnboardingProps
+}
+
+export interface OnboardingProps {
+  recoveringFromStoreWipe: boolean
+  choseToRestoreAccount: boolean | undefined
+  supportedBiometryType: BIOMETRY_TYPE | null
+  skipVerification: boolean
+  numberAlreadyVerifiedCentrally: boolean
+}
+
+export function onboardingPropsSelector(state: RootState): OnboardingProps {
+  const recoveringFromStoreWipe = recoveringFromStoreWipeSelector(state)
+  const choseToRestoreAccount = choseToRestoreAccountSelector(state)
+  const supportedBiometryType = supportedBiometryTypeSelector(state)
+  const skipVerification = skipVerificationSelector(state)
+  const numberAlreadyVerifiedCentrally = numberVerifiedCentrallySelector(state)
+
+  return {
+    recoveringFromStoreWipe,
+    choseToRestoreAccount,
+    supportedBiometryType,
+    skipVerification,
+    numberAlreadyVerifiedCentrally,
+  }
+}
+
+/**
+ * Traverses through the directed graph of onboarding navigate, navigateClearingStack, and navigateHome calls
+ * and count the number of screens until the given screen and the total number
+ */
+export function getOnboardingStepValues(screen: Screens, onboardingProps: OnboardingProps) {
+  let stepCounter = 1 // will increment this up to the onboarding step the user is on
+  let totalCounter = 1
+  let reachedScreeen = false // tracks whether we have reached the screen the user is on in onboarding, and we can stop incrementing stepCounter
+  let currentScreen: Screens = FIRST_ONBOARDING_SCREEN // pointer that we will update when simulating navigation through the onboarding screens to calculate "step" and "totalSteps"
+
+  const navigate = (s: Screens) => {
```

Comment:
nit: perhaps adjust naming on `navigate` and `navigateHome` to clarify that they're dummy functions? Without looking at the definitions, I would think that `navigate` and `navigateHome` actually do the thing they claim to.